### PR TITLE
CML Dvc Workflow

### PR DESCRIPTION
**Pull request by dragarok**

### CODE REVIEW

File: ContinuousML/NQ_DR_4_10_20_Ideal_27/1_Label_And_Feature_Workflow/Readme.md [hunk 1]
```diff
@@ -1,19 +1,21 @@
 # Feature Importance Readme
-- [ ] Build out dvc workflow for determining feature importance on Full_features.csv
-  - [ ] dvc.yaml
-    - [ ] dep: 
-      - [ ] ./src/Feature_Importance.py script for feature reduction
+- [-] Build out dvc workflow for determining feature importance on Full_features.csv
+  - [-] dvc.yaml
```
Comment (Roger-Parkinson-EHP) [inline ContinuousML/NQ_DR_4_10_20_Ideal_27/1_Label_And_Feature_Workflow/Readme.md]: replace withj x instead of a dash

File: ContinuousML/NQ_DR_4_10_20_Ideal_27/2_Training_Workflow/dvc.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+stages:
+  training:
+    cmd: python ../../../src/Model_Training/Model_Training.py
+    deps:
+    - ../../../src/Model_Training/Model_Training.py
+    - ../1_Label_And_Feature_Workflow/Full_Features.csv
```
Comment (Roger-Parkinson-EHP) [inline ContinuousML/NQ_DR_4_10_20_Ideal_27/2_Training_Workflow/dvc.yaml:6]: - Replace Full_features.csv with Reduced_Features.csv
- Temporary only during dev:  
  - Reduced_Features.csv can be produced from Full_features with column filtering Provided by Selected_features.csv

File: ContinuousML/NQ_DR_4_10_20_Ideal_27/2_Training_Workflow/dvc.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    - Selected_Features.csv
+    outs:
+    - keras-model/
```
Comment (Roger-Parkinson-EHP) [inline ContinuousML/NQ_DR_4_10_20_Ideal_27/2_Training_Workflow/dvc.yaml:9]: - Embed in 2_*
  - DVC add

File: src/Model_Training/Model_Training.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        None"""
+    gcp_bucket = 'tfc-cml'
+    BATCH_SIZE = 32
+
+    df = pd.read_csv("../1_Label_And_Feature_Workflow/Full_Features.csv")
+
+    features_df = df.drop(['Label', 'Bar'], axis=1)
+    labels_df = df['Label']
+
+    sel_features_df = pd.read_csv('Selected_Features.csv')
```
Comment (Roger-Parkinson-EHP) [inline src/Model_Training/Model_Training.py:26]: Change name to Reduced_Features.csv

File: src/Model_Training/Model_Training.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        # tf.keras.callbacks.TensorBoard(log_dir=tensorboard_path, histogram_freq=1),
+        # ModelCheckpoint will save models after each epoch for retrieval later.
+        tf.keras.callbacks.ModelCheckpoint(checkpoint_path),
+        # EarlyStopping will terminate training when val_loss ceases to improve.
+        # tf.keras.callbacks.EarlyStopping(monitor="val_loss", patience=3),
+    ]
+
+    model.compile(loss='sparse_categorical_crossentropy',
+                  optimizer=tf.keras.optimizers.Adam(),
+                  metrics=['accuracy'])
```
Comment (Roger-Parkinson-EHP) [inline src/Model_Training/Model_Training.py:66]: - I think I would prefer F1 as metric

File: src/Model_Training/Model_Training.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    # model.fit(train_dataset, callbacks=callbacks, epochs=1)
+    model.fit(train_dataset, epochs=1)
+
+    MODEL_PATH = "../2_Training_Workflow/keras-model"
```
Comment (Roger-Parkinson-EHP) [inline src/Model_Training/Model_Training.py:71]: - I want to model to be saved in ONNX format

File: src/Model_Training/Model_Training.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    # SAVE_PATH = os.path.join("gs://", gcp_bucket, MODEL_PATH)
+    SAVE_PATH = MODEL_PATH
+    model.save(SAVE_PATH)
+
```
Comment (Roger-Parkinson-EHP) [inline src/Model_Training/Model_Training.py:75]: - Need to add DVC add for the trained model
- Need to use CML to add Visuals to PR and Markdown

File: .github/workflows/cml.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+          with:
+            service_account_key: ${{ secrets.GCP_SA_KEY }}
+            project_id: ${{ secrets.GCS_PROJECT }}
+            export_default_credentials: true
+        - name: Select Features
+          env:
+            REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          run: |
+            # Your ML workflow goes here
+            pip install -r src/requirements.txt
```
Comment (Roger-Parkinson-EHP) [inline .github/workflows/cml.yaml:19]: - Roger thinks this should be just DVC repro
